src/components/navigation/Navigation.js

```diff
@@ -4,17 +4,66 @@ import { Link, NavLink } from 'react-router-dom';
 
 const links = [
   { label: 'Home', path: '', icon: 'home', auth: false },
-  { label: 'Home', path: '', icon: 'home', auth: true },
+  { label: 'Home-auth', path: '', icon: 'home', auth: true },
   { label: 'Shop', path: 'categories', icon: 'list-alt', auth: false },
   { label: 'Categories', path: 'categories', icon: 'list-alt', auth: true },
   { label: 'Products', path: 'products', icon: 'shopping-bag', auth: true },
   { label: 'Contacts', path: 'contacts', icon: 'map-signs' }
 ];
 
 export const Navigation = ({ user, onLogout }) => {
+  const mainNavLinks = user ?
+    links.filter(item => {
+      if (item.auth === true || item.auth === undefined) {
+        return (
+          <li key={item.label}>
+            <NavLink to={`/${item.path}`}>{item.label}</NavLink>
+          </li>
+        )
+      }
+    })
+  :
+    links.filter(item => {
+      if (item.auth === false || item.auth === undefined) {
+        return (
+          <li key={item.label}>
+            <NavLink to={`/${item.path}`}>{item.label}</NavLink>
+          </li>
+        )
+      }
+    })
```

Comment:
1) lots of repeated code and the only differences are 
```
if (item.auth === true
if (item.auth === false
```
2) please, check how Array.prototype.filter works (http://learn.javascript.ru/array-iteration#filter)